src/main/java/com/somesec/crypto/encrypt/AESEncryption.java

```diff
@@ -46,6 +50,27 @@ public byte[] encrypt(byte[] payload, Key key) {
         }
     }
 
+    @Override
+    public void encrypt(InputStream plainText, OutputStream cipherText, Key key) {
+        try (plainText; cipherText) {
+            Cipher cipher = Cipher.getInstance(resolver.getConfig(DefaultConfig.AES_CIPHER_NAME), BouncyCastleProvider.PROVIDER_NAME);
+            SecureRandom random = SecureRandom.getInstanceStrong();
+            final byte[] nonce = new byte[(int) resolver.getConfig(DefaultConfig.AES_GCM_NONCE_LENGTH)];
+            random.nextBytes(nonce);
+            GCMParameterSpec spec = new GCMParameterSpec((int) resolver.getConfig(DefaultConfig.AES_GCM_TAG_LENGTH_BYTE) * ((int) resolver.getConfig(DefaultConfig.BIT_IN_A_BYTE)), nonce);
+            cipher.init(Cipher.ENCRYPT_MODE, key, spec);
+            if (plainText.available() > 0) {
+                cipherText.write(nonce);
+            }
+            while (plainText.available() > 0) {
+                cipherText.write(cipher.update(plainText.readNBytes(resolver.getConfig(DefaultConfig.MEGA_BYTE))));
+            }
+            cipherText.write(cipher.doFinal());
```

Comment:
what do you think if we also call #flush() before closing the streams to ensure all last bytes are written?

Reply:
Yes thats definitely a good idea.